third_party/terraform/utils/test_utils.go

```diff
@@ -66,3 +69,40 @@ func toBool(attribute string) (bool, error) {
 	}
 	return strconv.ParseBool(attribute)
 }
+
+func checkDataSourceStateMatchesResourceState(dataSourceName, resourceName string) func(*terraform.State) error {
+	return checkDataSourceStateMatchesResourceStateWithIgnores(dataSourceName, resourceName, map[string]struct{}{})
+}
+
+func checkDataSourceStateMatchesResourceStateWithIgnores(dataSourceName, resourceName string, ignoreFields map[string]struct{}) func(*terraform.State) error {
+	return func(s *terraform.State) error {
+		ds, ok := s.RootModule().Resources[dataSourceName]
+		if !ok {
+			return fmt.Errorf("root module has no resource called %s", dataSourceName)
+		}
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("can't find %s in state", resourceName)
+		}
+
+		dsAttr := ds.Primary.Attributes
+		rsAttr := rs.Primary.Attributes
+
+		errMsg := ""
+		for k, attr := range rsAttr {
```

Comment:
Can you comment with why it's safe to iterate only over the resource fields?

Reply:
Done